docs/1-Applying/when_to_apply.md

```diff
@@ -1,9 +1,23 @@
+Note that the intern class _could_ be full. It won't necessarily be. If, for any reason, you are applying after New Years, there are still plenty of chances of getting an internship. We only mean that you could lose out on a few opportunities. Although getting an internship at top companies is competitive, there are probably more jobs than applicants in the industry overall. We're lucky to be in tech because of how many job opportunities there are!
+
+How long a company will accept summer internship application is very circumstantial and will change from year to year. There's no general rule that will always be true. However, understanding the situation of different companies could guide you in targeting your efforts:
+
+- *Medium and large tech companies*. Well-known tech companies such as Google, Microsoft, Dropbox, Airbnb, etc have dedicated recruiting teams, with each individual recruiters assigned to different universities. Due to their large size, their dedicated internship programs, and the large number of applicants, they need to plan well in advance. This often means that they will have a certain number of intern slots (rough of exact), sometimes split amongst different universities. They may leave a few spare slots for late applicants but don't count on it. You should probably apply to those when they start coming to campus in September/October. They tend to be in a hurry to recruit since they compete against each other for the best students.
+- *Startups* (check out [AngelList](https://angel.co/)). They could be the opposite. Their business is very uncertain and their hiring capacity, needs, projects, business goals, etc, tend to change randomly and quickly. They might only be able to know for sure if they can afford an intern when it's a month or two before the internship would start. While not all startups hire interns, it's worth reaching out to them. We often hear stories of students getting a job at a cool startup at the last minute, right after a slot opened up because the startup got funding (also at the last minute).
+- *Smaller companies*. There are many respectable companies, tech or non-tech, that only have a few dozen employees and can only hire a few interns. They don't have the resources to go campus recruiting and prefer to avoid competing head-on with big-name companies. You may have to send out a lot of emails but many of them will hire later. A lot of these companies might be local, so it's useful to search for tech companies in particular cities.
+- *Non-tech companies*. Similarly, there are many companies that hire interns for programming positions but don't have the prestige of large tech companies. Think banks for example. They might even have a lot of internships slots, but they won't get filled as quickly. Keep in mind, just because a company isn't famous in tech doesn't mean it doesn't have good learning opportunities!
```

Comment:
> just because a company isn't famous in tech doesn't mean it doesn't have good learning opportunities!

omg I want to make a whole file just about this lol